services/s3/errors.go

```diff
@@ -21,3 +21,14 @@ func NoSuchBucket(bucket string) *awserrors.Error {
 		},
 	}
 }
+
+func VersionConflict(key string) *awserrors.Error {
+	return &awserrors.Error{
+		Code: 409,
```

Comment:
is this correct? My reading of docs is that this should be a 412, but I'm not sure what would cause a 409...
<img width="936" alt="image" src="https://github.com/user-attachments/assets/f448a93f-1c2a-41fc-a50b-120d63ca9162">

Reply:
oh, 409 is in case of a delete racing with the Put, see https://docs.aws.amazon.com/AmazonS3/latest/userguide/conditional-requests.html#conditional-error-response